Documentation/feature-band-source-building.md

```diff
@@ -0,0 +1,825 @@
+
+        Prev_SDK -.-> Build
+        Prev_Art -.->|PSB| Build
+        Build --> New_SDK
+        Build --> New_Art
+    end
+    
+    classDef prev1xx fill:#db9b9b,stroke:#666,stroke-width:2px,color:#000
+    classDef curr1xx fill:#5bb75b,stroke:#666,stroke-width:2px,color:#000
+    class Msft prev1xx
+    class SB curr1xx
+```
+
+### Scenario 3: 2xx Band Initial Release (N.0.200)
+
+Initial release of the 2xx band.
+
+Required inputs:
+
+- source-built artifacts of the current 1xx release
+- source-built SDK artifacts from the previous 1xx release
+
+```bash
+git clone -b <2xx-release-branch> https://github.com/dotnet/dotnet.git
+cd dotnet
+
+# Build the SDK
+./build.sh --source-only \
+  --with-sdk /path/to/previous-source-built-1xx/sdk \
+  --with-packages /path/to/previous-source-built-1xx/artifacts \
+  --with-shared-components /path/to/current-source-built-1xx/artifacts
+
+# Final source-built outputs available in artifacts/x64/Release/
+```
+
+```mermaid
+flowchart LR
+    subgraph SBPrev["Previous 1xx SB Release"]
+        Prev_1xx_SDK[Source-built 1xx SDK]
+        Prev_1xx_Art[Source-built 1xx artifacts]
+    end
+    
+    subgraph SBCurrent1xx["Current SB 1xx Release"]
+        Curr_1xx_Art[Current source-built<br/>1xx Artifacts]
+    end
+    
+    subgraph SBCurrent2xx["Current SB 2xx Release"]
+        Build((Build Process))
+        SDK_2xx[New 2xx SDK]
+        Art_2xx[New 2xx Artifacts]
+
+        Prev_1xx_SDK -.-> Build
+        Prev_1xx_Art -.->|PSB| Build
+        Curr_1xx_Art -.->|shared components| Build
+        Build --> SDK_2xx
+        Build --> Art_2xx
+    end
+    
+    classDef prev1xx fill:#db9b9b,stroke:#666,stroke-width:2px,color:#000
+    classDef curr1xx fill:#5bb75b,stroke:#666,stroke-width:2px,color:#000
+    classDef curr2xx fill:#3da53d,stroke:#666,stroke-width:2px,color:#000
+    class SBPrev prev1xx
+    class SBCurrent1xx curr1xx
+    class SBCurrent2xx curr2xx
+```
+
+### Scenario 4: 2xx Band Bootstrap (N.0.200+)
+
+For 2xx releases that require bootstrap (two-stage process).
+
+Required inputs:
+
+- source-built artifacts of the current 1xx release
+
+```bash
+git clone -b <2xx-release-branch> https://github.com/dotnet/dotnet.git
+cd dotnet
+
+# ===
+# Stage 1: Build with Microsoft artifacts
+# ===
+
+# Downloads Microsoft-built artifacts and SDK
+./prep-source-build.sh
+
+# Build the SDK referencing assets from current 1xx release
+./build.sh --source-only --with-shared-components /path/to/source-built-1xx/artifacts
+
+# Extract and store the built 2xx SDK and artifacts
+tar -ozxf artifacts/assets/Release/dotnet-sdk-*-tar.gz \
+  -C /tmp/dotnet/sdk
+tar -ozxf artifacts/assets/Release/Private.SourceBuilt.Artifacts.*.tar.gz \
+  -C /tmp/dotnet/artifacts
+
+# ===
+# Stage 2: Rebuild using stage 1 outputs
+# ===
+
+./build.sh --source-only \
+  --with-sdk /tmp/dotnet/sdk \
+  --with-packages /tmp/dotnet/artifacts
+
+# Final source-built outputs available in artifacts/x64/Release/
+```
+
+```mermaid
```

Comment:
I think it would be good to include 2xx in the Microsoft-built SDK and artifacts nodes.

Reply:
I intentionally omitted that. Technically, it would be 1xx if the current version of 2xx feature band was 200. It would be 2xx if the current version was 201+. So I left it out to be more general. Given that this is a bootstrapping scenario, it doesn't really matter to the distro maintainer where the Microsoft artifacts came from. I do point out this difference in the `Input Artifacts Summary` section. But I didn't really feel it was necessary in the scenario since it doesn't matter to the person running the steps.